src/main.rs

```diff
@@ -50,10 +50,66 @@ fn main() {
     }
 }
 
+fn build_cache_without_training(cl: clap::ArgMatches) -> Result<(), Box<dyn Error>> {
+    /*! A method that enables creating the cache file without training the first model instance.
+    This is done in order to reduce building time of the cache and running the first model instance multi threaded. */
+    // We'll parse once the command line into cl and then different objects will examine it
+    let input_filename = cl.value_of("data").expect("--data expected");
+    let vw_namespace_map_filepath = Path::new(input_filename)
+        .parent()
+        .expect("Couldn't access path given by --data")
+        .join("vw_namespace_map.csv");
+    let vw: vwmap::VwNamespaceMap;
+    vw = vwmap::VwNamespaceMap::new_from_csv_filepath(vw_namespace_map_filepath)?;
+    let mut cache = cache::RecordCache::new(input_filename, cl.is_present("cache"), &vw);
+    let input = File::open(input_filename)?;
+    let mut aa;
+    let mut bb;
```

Comment:
the following section is mostly a duplication of the training process further down the file.
is it possible to avoid the duplication and instead set a flag and use it in the normal flow later?

Reply:
at first i thought about it but i didnt want to go through the entire process of reading and parsing arguments if i only want to build the cache and exit

Reply:
you can skip most of them if you are in "cache only" mode..
the thing is - if we leave it as is (with the code duplication), it means that in the future if anyone changes the cache flow they have to change both flows... and we have different code generating the "cache only" produced cache vs. the "not cache only" cache.